logstash-core/spec/logstash/instrument/periodic_poller/dlq_spec.rb

```diff
@@ -0,0 +1,42 @@
+# encoding: utf-8
+require "spec_helper"
+require "logstash/instrument/periodic_poller/dlq"
+require "logstash/instrument/collector"
+require "logstash/environment"
+require "logstash/agent"
+
+describe LogStash::Instrument::PeriodicPoller::DeadLetterQueue do
+  subject { LogStash::Instrument::PeriodicPoller::DeadLetterQueue }
+
+  let(:metric) { LogStash::Instrument::Metric.new(LogStash::Instrument::Collector.new) }
+  let(:agent) { double("agent")}
+  let(:pipeline) { double("pipeline")}
+  let(:pipelines) { { :main => pipeline } }
+  let(:options) { {} }
+  subject(:dlq) { described_class.new(metric, agent, options) }
+
+  it "should initialize cleanly" do
+    expect { dlq }.not_to raise_error
+  end
+
+
+
+  describe 'collect' do
+    subject(:collect) { dlq.collect }
```

Comment:
If you aren't going to invoke `subject` in the tests, I don't think you need this.

Later down in the tests (line 33, etc) you can do:

```
expect { dlq.collect }.not_to  raise_error
```

etc.